### PR TITLE
First draft of vespa vector search

### PR DESCRIPTION
**Pull request by jobergum**
Kristian - could you help iterate on this PR to get more examples/improvements? 

- Maybe connect with pyvespa and indexing and searching some vector data
- Add readme testing/ci-cd

### CODE REVIEW

**Comment by kkraune** [comment]
I will make a few comments on the readme

**Review by bratseth — APPROVED** [review]
💯

File: README.md [hunk 1]
```diff
@@ -1,2 +1,185 @@
-# vector-search
-Vector Search Sample Application for Vespa Cloud 
+<!-- Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.-->
+![Vespa logo](https://vespa.ai/assets/vespa-logo-color.png)
```
Comment (kkraune) [inline README.md]:
```suggestion
![Vespa Cloud logo](https://cloud.vespa.ai/assets/logos/vespa-cloud-logo-full-black.png)
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,185 @@
...
+no credit card required. 
+
+## Clone this repo 
+<pre>
+git clone --depth 1 git@github.com:vespa-cloud/vector-search.git && cd vector-search
+</pre>
+
+## Install Vespa-CLI
+Install the [Vespa-CLI](https://docs.vespa.ai/en/vespa-cli.html) which is the official command-line
+client for interacting with Vespa.
```
Comment (kkraune) [inline README.md]:
```suggestion
client for interacting with Vespa:
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,185 @@
...
+brew install vespa-cli 
+</pre>
+
+You can also download [Vespa CLI](https://github.com/vespa-engine/vespa/releases) 
+for Windows, Linux and macOS.
+
+## Configure Vespa-CLI 
+Replace `<tenant-name>` with your Vespa Cloud tenant name. 
+In this case, the application name used is `vector-search` and instance
+is `default`.
```
Comment (kkraune) [inline README.md]:
```suggestion
In this case, the application name used is `vector-search` and instance is `default`:
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,185 @@
...
+<pre>
+vespa auth cert
+</pre>
+
+Read more about how Vespa Cloud keeps your data safe and private at rest and in transit 
+in the [Vespa Cloud Security Guide](https://cloud.vespa.ai/en/security/guide).
+
+## Configure Vector Schema 
+Now the app is ready to be deployed. The [vector schema](schemas/vector.sd)
+is configured with `768` dimension using `float` precision.
```
Comment (kkraune) [inline README.md]:
```suggestion
is configured with `768` dimensions using `float` precision. 
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,185 @@
...
+* Change vector precision type (Default `float`) - Choose between `int8`, `bfloat16` or `float`.
+* Change [distance-metric](https://docs.vespa.ai/en/reference/schema-reference.html#distance-metric) 
+(Default `euclidean`) - Also supported `angular`, `innerproduct` and `hamming`.
+
+
+## Deploy to dev environment 
+Vespa Cloud supports multiple different [environments](https://cloud.vespa.ai/en/reference/environments) and
+the following guides you through the process of deploying to `dev` for developing and testing
+of your vector search use case, to deploying to `perf` for performance validation and benchmarking, and finally to
+`prod` for high availability production serving.
```
Comment (kkraune) [inline README.md]:
```suggestion
Vespa Cloud supports multiple different [environments](https://cloud.vespa.ai/en/reference/environments).
The following guides you through:
* Deploying to `dev` for developing and testing of your vector search use case
* Deploying to `perf` for performance validation and benchmarking
* Deploying to `prod` for high availability production serving
```

File: README.md [hunk 6]
```diff
@@ -1,2 +1,185 @@
...
+The first deployment to dev environment takes about 12 minutes for provisioning resources and 
+setting up certificates. 
+
+## Deploy to perf environment
+
+The perf zone is used for [benchmarking](https://cloud.vespa.ai/en/benchmarking) and performance testing. 
+It uses the same resource specification as in production, except
+for redundancy. 
+
+Deploy app to `perf` by using the `--zone` parameter
```
Comment (kkraune) [inline README.md]:
```suggestion
Deploy app to `perf` by using the `--zone` parameter:
```

File: README.md [hunk 7]
```diff
@@ -1,2 +1,185 @@
...
+We recommend deploying using CI/CD, for example deploying to Vespa Cloud using GitHub Actions.  
+
+## Vespa Cloud - Vector Search Price Examples
+
+[Vespa Cloud pricing](https://cloud.vespa.ai/pricing) is simple and transparent. 
+All customers receive all features and services, and is charged a fee proportional to the resources the application uses. 
+
+The production env configuration in [services.xml](services.xml) specifies the following resources:
+<pre>
+ &lt;nodes deploy:environment="prod" count="2" groups="2"&gt;
```
Comment (kkraune) [inline README.md]:
```suggestion
&lt;nodes deploy:environment="prod" count="2" groups="2"&gt;
```

File: README.md [hunk 8]
```diff
@@ -1,2 +1,185 @@
...
+writes per second depends on [vector search parameters](https://docs.vespa.ai/en/approximate-nn-hnsw.html).
+
+Vespa Cloud sizing experts can assist in finding the most cost efficient resource specification matching your vector search 
+use case. Sizing and cost estimation uses samples of your data using the `perf` environment. 
+
+Vespa Cloud also supports [auto-scaling](https://cloud.vespa.ai/en/autoscaling), in this example scaling 
+number of groups (replicas) to handle query traffic changes.
+
+<pre>
+ &lt;nodes deploy:environment="prod" count="[2,4]" groups="[2,4]"&gt;
```
Comment (kkraune) [inline README.md]:
```suggestion
&lt;nodes deploy:environment="prod" count="[2,4]" groups="[2,4]"&gt;
```

**Review by kkraune — APPROVED** [review]
Great! The usual nits.

**Comment by kkraune** [comment]
I suggest we modify https://pyvespa.readthedocs.io/en/latest/deploy-vespa-cloud.html to using this as the sample config, @thigm85

**Pull request merged**